### PR TITLE
platform: ace: notifying about idle thread readiness

### PR DESCRIPTION
**Pull request by tmleman**
Informing the primary core that the Idle thread on secondary core is ready. During the D3 exit flow thread is not initialize again, but restored from previously saved context.

Requires:

- [x] https://github.com/zephyrproject-rtos/zephyr/pull/55182

### CODE REVIEW

File: src/platform/intel/ace/platform.c
Comment (lgirdwood) [inline src/platform/intel/ace/platform.c:119]: Is long term switch over to AMS here ?

**Review by aborisovich — CHANGES_REQUESTED** [review]

File: src/platform/intel/ace/platform.c
Comment (tmleman) [reply src/platform/intel/ace/platform.c:128]: No, it will stay as it is (if nothing changes).

File: zephyr/lib/cpu.c
Comment (lgirdwood) [reply zephyr/lib/cpu.c:33]: We could save a function call and static inline this public API too.

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

File: zephyr/lib/cpu.c
Comment (marcinszkudlinski) [inline zephyr/lib/cpu.c:87]: why we need to check zephyr version? We won't ever go back to older zephyr releases.

Comment (paulstelian97) [reply zephyr/lib/cpu.c:87]: Are there plans to integrate some code from here into Zephyr itself as opposed to it staying in SOF? If so, then these checks are still worth it.

Comment (lgirdwood) [reply zephyr/lib/cpu.c:96]: @paulstelian97 yes, that the direction, IIUC there are still some dependencies to upstream.
I agree though, we dont need the version check as west manifest should allow us to specify supported Zephyr commit.
@tmleman does this PR need a west update ?

Comment (tmleman) [reply zephyr/lib/cpu.c:96]: > does this PR need a west update ?

@lgirdwood Yes, at first I need to merge zephyr PR and after that this can be merged with zephyr update. But regardless of that, everyone can review and approve.

**Review by aborisovich — APPROVED** [review]

**Comment by lgirdwood** [comment]
@tmleman any update ?

**Comment by tmleman** [comment]
> any update ?

@lgirdwood I'm still waiting for the merge on the zephyr side

File: zephyr/lib/cpu.c
Comment (kv2019i) [inline zephyr/lib/cpu.c:102]: Given this is an open-coded  of z_smp_start_cpu() (see comment above), I don't fully understand why we have a different implementation here, and if this is correct, why this is not needed in the Zephyr kernel/smp.c version. Wouldn't the same problem be there? Or is this specific to a configuration where state is saved to persistant memory? If yes, this would need to be ifdef'ed out so this applies only to applicable platforms.

**Review by kv2019i — CHANGES_REQUESTED** [review]
Now I see in second commit this is related to state-save. This should be somehow conditional on the feature being used, given the generic code is used for all platform/vendors.

File: zephyr/lib/cpu.c
Comment (kv2019i) [reply zephyr/lib/cpu.c:102]: @nashif @teburd @dcpleung @andyross  ^^ trying to understand why z_smp_start_cpu() does not need this?

**Comment by kv2019i** [comment]
Usual PM fail in https://sof-ci.01.org/sofpr/PR7174/build6662/devicetest/index.html (but passes on at least one device) and https://sof-ci.01.org/sofpr/PR7174/build6663/devicetest/index.html has one fail on system PM.

Proceeding with the fix. Thanks @tmleman for filing the issue.

**Pull request merged**
